pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/timeseries/serde/TimeSeriesBlockSerde.java

```diff
@@ -104,7 +110,21 @@ public static TimeSeriesBlock deserializeTimeSeriesBlock(ByteBuffer readOnlyByte
       long seriesId = Long.parseLong(timeSeries.getId());
       seriesMap.computeIfAbsent(seriesId, x -> new ArrayList<>()).add(timeSeries);
     }
-    return new TimeSeriesBlock(timeBuckets, seriesMap, metadataMap);
+    TimeSeriesBlock block = new TimeSeriesBlock(timeBuckets, seriesMap, metadataMap);
+    if (metadataMap.containsKey(EXCEPTIONS_METADATA_KEY)) {
+      String exceptionsJson = metadataMap.get(EXCEPTIONS_METADATA_KEY);
+      try {
+        List<Map<String, Object>> exceptionsList = JsonUtils.stringToObject(exceptionsJson, List.class);
+        for (Map<String, Object> exceptionData : exceptionsList) {
+          int errorCode = ((Number) exceptionData.get(ERROR_CODE)).intValue();
+          String message = (String) exceptionData.get(MESSAGE);
+          block.addToExceptions(new QueryException(QueryErrorCode.fromErrorCode(errorCode), message));
+        }
+      } catch (Exception e) {
+        throw new IOException("Failed to deserialize exceptions from metadata", e);
```

Comment:
Hmm let's change the signature to throw the checked "Exception"? We shouldn't ideally be wrapping into IOException ourselves.